gonzo/clouds/compute.py

```diff
@@ -278,6 +286,65 @@ def create_volume(self, instance, vol_name,
             ex_volume_type=vol_type,
         )
 
+    def create_if_not_exist_security_group(self, group_name, subnet_id=None):
+        if subnet_id is not None:
+            subnet = self.compute_session.ex_list_subnets([subnet_id])[0]
```

Comment:
as the `subnet_id` is entered through the command line (i think) you might get bad IDs here which you could handle?